libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -89,8 +91,13 @@ AP_BattMonitor_Analog::read()
         _state.healthy &= _curr_pin_analog_source->set_pin(_curr_pin);
 
         // read current
-        _state.current_amps = (_curr_pin_analog_source->voltage_average() - _curr_amp_offset) * _curr_amp_per_volt;
-
+#if !APM_BUILD_TYPE(APM_BUILD_AP_Periph)
+        _state.current_amps = (gcs().get_hud_throttle() * 0.01)* _curr_amp_per_volt - _curr_amp_offset ;
+#endif // no synthetic current sensor in AP_Periph builds
+        if (!(uint32_t(_params._options.get()) & uint32_t(AP_BattMonitor_Params::Options::Synthetic_Current_Sensor))) {
+            _state.current_amps = (_curr_pin_analog_source->voltage_average() - _curr_amp_offset) * _curr_amp_per_volt;            
+        } 
+        
```

Comment:
```suggestion
#if !APM_BUILD_TYPE(APM_BUILD_AP_Periph)
        if (uint32_t(_params._options.get()) & uint32_t(AP_BattMonitor_Params::Options::Synthetic_Current_Sensor)) {
_curr_amp_per_volt;
            _state.current_amps = (gcs().get_hud_throttle() * 0.01)* _curr_amp_per_volt - _curr_amp_offset ;
        } else
#endif // !APM_BUILD_TYPE(APM_BUILD_AP_Periph)
       {
            _state.current_amps = (_curr_pin_analog_source->voltage_average() - _curr_amp_offset) * 
        }
        
```

Let's flip the logic around here, it's easier to read, and saves us from calling stuff we don't need if you have an actual current sensor.

Reply:
@WickedShell I did it that way to save flash, which it does vs the way you show, and changing the sign on the offset math is fine by me and makes the sign of the BATT_AMP_OFFSET match how you would set the idling current for the system....if the group wants to spend the extra flash, I have no objection..I did it that way first then thought about how I could save flash

@IamPete1 many of the boards that would use this do not have scripting capability

Reply:
If a single if statement is to much flash then we shouldn't be adding this. There really isn't a good reason to go through the hassle of getting the throttle if it's not needed. It's far more indirect in some cases then you would have expected. It's also easier to read the code if we don't do that.

You have to handle the negative for reverse thrust at a minimum.

When you say handle idle current, do you actually want to just add the offset value to account for the idle current? That actually does make sense to me, but in that case why not add it (IE a offset of 0.5 for a half amp idle makes more sense to me from a user perspective then -0.5. But that might just be me :) )

Reply:
changed back to if/else and made offset add for idling system currents....